h/groups/models.py

```diff
@@ -58,6 +59,34 @@ def slug(self):
     def __repr__(self):
         return '<Group: %s>' % self.slug
 
+    def documents(self, limit=25):
+        """
+        Return this group's most recently annotated documents.
+
+        Only returns documents that have shared annotations in this group,
+        not documents that only have private annotations in the group.
+
+        """
+        documents = []
+        annotations = (
+            sa.orm.object_session(self).query(models.Annotation)
+            .options(sa.orm.load_only('id'))
+            .options(sa.orm.joinedload('document')
+                     .joinedload('document_uris'))
+            .options(sa.orm.joinedload('document')
+                     .joinedload('meta'))
+            .filter_by(groupid=self.pubid, shared=True)
+            .order_by(models.Annotation.updated.desc())
+            .limit(1000)
+            .all())
+        for annotation in annotations:
+            if annotation.document not in documents:
```

Comment:
Hmm, ok. There are ways around that (use a set of ids, say, and a list) but it's only 25 by default so I guess it'll be fine.